scipy/sparse/linalg/dsolve/linsolve.py

```diff
@@ -294,7 +296,9 @@ def splu(A, permc_spec=None, diag_pivot_thresh=None,
         A = csc_matrix(A)
         warn('splu requires CSC matrix format', SparseEfficiencyWarning)
 
-    A.sort_indices()
+    # sum duplicates for non-canonical format
+    if not A.has_canonical_format:
+        A.sum_duplicates()
```

Comment:
In each of these cases you can just call `A.sum_duplicates()`. It checks if it's necessary internally before doing any work.